### ISSUE TITLE
AWS S3 FS with Fuse

### ISSUE THREAD

**Issue by janoist1** [ai-dock/comfyui#33]
Hi,

I've managed to run this on an AWS EC2 with T4 GPU.
I'd like to launch a more powerful instance for quicker video generation, however I'd like to maintain 1 single setup at the lowest cost possible.
First I came out with an idea of having all the workspace stuff in S3 mounted into the instances using Fuse S3 filesystem. I begun to implement it:

1. mount the S3 bucket
2. rsync workspace to the bucket
3. removed the original workspace
4. symlink the S3 mount point to where the original workspace was

ComfyUI kept working, I was very happy.

The problems arisen when I re-created the instance, mounted the S3 bucket, then docker run pointing workspace to the S3 bucket where the workspace had been created previously. Here I got ACL issues first, then after switching it off via env var, remove container and re-execute docker run, I realised that a new directory called ComfyUI-link was created - that might have been due to disabling `WORKSPACE_SYNC`. At this point I decided to turn to the github issue page.

Question is, is it possible to somehow use an S3 bucket for the ComfyUI config including all the models and stuff? What to be taken into account? Why there is a ComfyUI-link directory?

Thank you!

**Comment by robballantyne** [comment]
You could follow https://aws.amazon.com/blogs/storage/mounting-amazon-s3-to-an-amazon-ec2-instance-using-a-private-connection-to-s3-file-gateway/ but I would not expect performance to be as good as standard block storage. You should mount at /workspace 

ComfyUI link is created (pointed to /opt/ComfyUI) if you have a ComfyUI directory in the $WORKSPACE directory but $WORKSPACE_SYNC disabled.

**Comment by janoist1** [comment]
Thanks for the answer. In the meantime I found a workaround that did not require me customise the docker image.
I ended up basically having a post-init script that logs into docker and mounts an S3 filesystem, symlinks some folders (models, custom_nodes, etc).
As a kind of proof of concept it has been working for 1-2 weeks now and I'm very happy with it. S3 is much cheaper than EBS and more importantly it can be mounted to multiple instances (with care), allowing me to spin up a much more powerful instance for demanding tasks.
On the flip side, it is generally slower, taking a bit more time to load models and the system itself, but that is something that can be lived with.
I personally would love to see it integrated into this image.
